src/components/pages/users.js

```diff
@@ -160,10 +162,57 @@ function Users({ initialState, PageTemplate }) {
     }
   };
 
+  const handleStoragePlanChange = async (user, newStoragePlanId) => {
+    const oldStorage = user.storage;
+    const newStorage = oldStorage
+      ? {
+        ...oldStorage,
+        plan: newStoragePlanId
+      }
+      : {
+        plan: newStoragePlanId,
+        usedStorageInBytes: 0,
+        reminders: []
+      };
+
+    setState(prevState => ({
+      ...prevState,
+      internalUsers: replaceUserById(prevState.internalUsers, { ...user, storage: newStorage }),
+      isSaving: true
+    }));
+
+    let finalStorage;
+    try {
+      finalStorage = await userApiClient.saveUserStoragePlan({ userId: user._id, storagePlanId: newStoragePlanId });
+    } catch (error) {
+      errorHelper.handleApiError({ error, logger, t });
+      finalStorage = oldStorage;
+    } finally {
+      setState(prevState => ({
+        ...prevState,
+        internalUsers: replaceUserById(prevState.internalUsers, { ...user, storage: finalStorage }),
```

Comment:
Small thing, I know this function was there before, but what about calling it `replaceUser`?
Usually this `byId` postfix requires passing the id, not the entity, which made me check if there is a bug because of the usage.